src/modules/convex/algo/igd.hpp

```diff
@@ -86,6 +101,32 @@ IGD<State, ConstState, Task>::merge(state_type &state,
 
 template <class State, class ConstState, class Task>
 void
+IGD<State, ConstState, Task>::mergeInPlace(state_type &state,
```

Comment:
I see. It makes sense. Both loss and gradient are kept for the purpose to decide the convergence and step size tuning, to the upper layer of internal code or end user. I'm sure what you do for that, I guess you still need to pass the gradient to the inter-iteration layer? or it's not necessary anymore for this approach?

I think code redundancy is not a huge deal in this short piece of code. The man issue is making people wonder about the difference between the functions and which one to use in the future. I would suggest at least some amount of comment to explain the difference before we have time to refactor it cleanly.

Great work!